### PR TITLE
Idol test suite

### PR DESCRIPTION
**Pull request by mkeeter**
I felt bad about adding `bool` arguments without corresponding tests, so here's a bunch of Idol unit tests!

The PR also adds a new test image for Gimletlet, since that's the most common board, and will probably be our CI target (eventually),

### CODE REVIEW

File: test/test-suite/src/main.rs [hunk 1]
```diff
@@ -11,9 +11,13 @@
 //!
 //! # The assistant
 //!
-//! This test suite uses a second task, the assistant, to test IPC and
-//! interactions. The assistant must be included in the image with the name
-//! `assist`, but its ID is immaterial.
+//! This test suite uses two other tasks to test IPC and interactions.
+//!
+//! The assistant, `test-assit`, tests raw test IPC and interactions.  It must
```
Comment (jgallagher) [inline test/test-suite/src/main.rs:16]:
```suggestion
//! The assistant, `test-assist`, tests raw test IPC and interactions.  It must
```

File: test/tests-gimletlet/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[package]
+edition = "2018"
+readme = "README.md"
+name = "tests-gemini-bu"
```
Comment (jgallagher) [inline test/tests-gimletlet/Cargo.toml:4]: Should this be `tests-gimletlet` (similarly the `[[bin]]` path below)?

**Comment by jgallagher** [comment]
LGTM! Tests pass for me on nucleo h743 and gimletlet.

File: test/tests-gimletlet/Cargo.toml (hunk 2 above)
Comment (mkeeter) [reply test/tests-gimletlet/Cargo.toml]: Too much copy-pasta, fixed!

**Review by flihp — COMMENTED** [review]
merged before review complete

**Pull request merged**